File: backend-engineering-work/src/main/java/com/example/backendengineeringwork/dto/car/CarPreview.java
package com.example.backendengineeringwork.dto.car;

import com.example.backendengineeringwork.dto.ImageCar.ResponseImageCarDto;
import com.example.backendengineeringwork.models.CarModel;

import java.util.List;

public record CarPreview(Long id,
                         String name,
                         Integer rentPrizePerDay,
                         Double acceleration,
                         Integer power,
                         Integer torque,
                         Integer seats,
                         String transmission,
                         Integer productionYear,
                         List<ResponseImageCarDto> imageCarList) {
}

File: frontend-engineering-work/src/app/models/car-preview.ts
import { ImageCar } from "./image-car";

export interface CarPreview {
  id?: number,
  name: string,
  rentPrizePerDay: number,
  acceleration: number,
  power: number,
  torque: number,
  seats: number,
  transmission: string,
  productionYear: number,
  imageCarList: ImageCar[],
}
